### PR TITLE
Edits to MP-QUIC and TCP over QUIC sections

### PR DESCRIPTION
**Pull request by mirjak**

### CODE REVIEW

File: draft-quic-atsss-reqs.mkd [hunk 1]
```diff
@@ -463,11 +463,21 @@ Network input can be especially beneficial for cases such as:
 
 This section elaborates how non-TCP traffic (UDP or a subset like QUIC) can to be encapsulated into a tunnel between the UE and the ATSSS UPF to enable so called ATSSS higher-layer steering for all traffic, not only TCP or QUIC. This section discusses to what extent QUIC can be used as a tunneling protocol for the ATSSS service and whether gaps are found.
 
+When tunneling non-TCP (e.g., UDP, plain IP) over QUIC the Unreliable Datagram Extension
+{{I-D.ietf-quic-datagram}} can be used. Each data packet would  transported unreliably as a datagram
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
{{I-D.ietf-quic-datagram}} can be used. Each data packet would be transported unreliably as a datagram
```

File: draft-quic-atsss-reqs.mkd [hunk 2]
```diff
@@ -463,11 +463,21 @@ Network input can be especially beneficial for cases such as:
...
+over the QUIC connection. Transporting these datagrams unreliably as
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
over a QUIC connection. Transporting these datagrams unreliably as
```

File: draft-quic-atsss-reqs.mkd [hunk 3]
```diff
@@ -463,11 +463,21 @@ Network input can be especially beneficial for cases such as:
...
+in IPSec or DTLS tunnels is especially important for flows that do not require reliable delivery
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
in IPsec or DTLS-based tunnels is especially important for flows that do not require reliable delivery 
```

File: draft-quic-atsss-reqs.mkd [hunk 4]
```diff
@@ -463,11 +463,21 @@ Network input can be especially beneficial for cases such as:
 
 This section elaborates how non-TCP traffic (UDP or a subset like QUIC) can to be encapsulated into a tunnel between the UE and the ATSSS UPF to enable so called ATSSS higher-layer steering for all traffic, not only TCP or QUIC. This section discusses to what extent QUIC can be used as a tunneling protocol for the ATSSS service and whether gaps are found.
 
+When tunneling non-TCP (e.g., UDP, plain IP) over QUIC the Unreliable Datagram Extension
+{{I-D.ietf-quic-datagram}} can be used. Each data packet would  transported unreliably as a datagram
+over the QUIC connection. Transporting these datagrams unreliably as
+in IPSec or DTLS tunnels is especially important for flows that do not require reliable delivery 
+and suffer from unnecessary delay when reliability is added. QUIC datagram are conngestion controlled,
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
and suffer from unnecessary delay when reliability is added. QUIC datagrams are congestion-controlled, 
```

File: draft-quic-atsss-reqs.mkd [hunk 5]
```diff
@@ -463,11 +463,21 @@ Network input can be especially beneficial for cases such as:
 
 This section elaborates how non-TCP traffic (UDP or a subset like QUIC) can to be encapsulated into a tunnel between the UE and the ATSSS UPF to enable so called ATSSS higher-layer steering for all traffic, not only TCP or QUIC. This section discusses to what extent QUIC can be used as a tunneling protocol for the ATSSS service and whether gaps are found.
 
+When tunneling non-TCP (e.g., UDP, plain IP) over QUIC the Unreliable Datagram Extension
+{{I-D.ietf-quic-datagram}} can be used. Each data packet would  transported unreliably as a datagram
+over the QUIC connection. Transporting these datagrams unreliably as
+in IPSec or DTLS tunnels is especially important for flows that do not require reliable delivery 
+and suffer from unnecessary delay when reliability is added. QUIC datagram are conngestion controlled, 
+however as the latency between the UE and the ATSSS proxy is small compared to the end-to-end latency
+this should not impact the end-to-end congestion control negative.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]: I don't parse this sentence.

File: draft-quic-atsss-reqs.mkd [hunk 6]
```diff
@@ -505,45 +515,45 @@ A third candidate solution is to leverage the ability of QUIC to support multipl
...
 a smartphone performs handovers, there are frequently short periods
-of time during which both networks are imperfect {{MPTester}}. During
-these periods, it is important to be able to use both networks
-simultaneously.
+of time during which both networks are imperfect {{MPTester}}. Using use both networks
+simultaneously during these periods, improves the user experience.
 
-Since there is a single Multipath QUIC session between the UE and the
-ATSSS UPF, load-balancers do not cause problems. 
+Further with splitting traffic can be better distributed based on available ressources
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
Furthermore, traffic can be better distributed among available paths based on available resources 
```

File: draft-quic-atsss-reqs.mkd [hunk 7]
```diff
@@ -562,21 +572,13 @@ TCP connection with a remote host. This is illustrated in {{fig-atssv2-tcp}}.
...
-datagrams. The MASQUE protocol uses the HTTP/3 CONNECT method to
-support TCP connections {{I-D.schinazi-masque-protocol}}
-and extend it with a UDPCONNECT method to
-support UDP {{I-D.schinazi-masque-connect-udp}}. A simpler protocol
-providing similar features is proposed in {{I-D.piraux-quic-tunnel}}{{I-D.piraux-quic-tunnel-tcp}}.
+Two application layer protocols are proposed to manage the mapping
+of TCP connections to QUIC streams as well as the transmission of UDP
+datagrams over QUIC datagrams. The to-be MASQUE working groups extends the HTTP/3 CONNECT method 
+with a UDPCONNECT method to use QUIC as a tunnel to proxy UDP traffic {{I-D.schinazi-masque-connect-udp}}.
+Another use case in scope for the groups is IP proxying which can be used for tunneling and forwarding of TCP connections.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]: which "groups"?

**Review by boucadair — CHANGES_REQUESTED** [review]

File: draft-quic-atsss-reqs.mkd (hunk 5 above)
Comment (mirjak) [reply draft-quic-atsss-reqs.mkd]: Any better now?

File: draft-quic-atsss-reqs.mkd [hunk 8]
```diff
@@ -463,11 +463,21 @@ Network input can be especially beneficial for cases such as:
...
 This section elaborates how non-TCP traffic (UDP or a subset like QUIC) can to be encapsulated into a tunnel between the UE and the ATSSS UPF to enable so called ATSSS higher-layer steering for all traffic, not only TCP or QUIC. This section discusses to what extent QUIC can be used as a tunneling protocol for the ATSSS service and whether gaps are found.
 
+When tunneling non-TCP (e.g., UDP, plain IP) over QUIC the Unreliable Datagram Extension
+{{I-D.ietf-quic-datagram}} can be used. Each data packet would be transported unreliably as a datagram
+over a QUIC connection. Transporting these datagrams unreliably as
+in IPsec or DTLS-based tunnels is especially important for flows that do not require reliable delivery 
+and suffer from unnecessary delay when reliability is added. QUIC datagrams are congestion-controlled, 
+however, as the latency between the UE and the ATSSS proxy is small compared to the end-to-end latency,
+having another local congestion control loops should not impact the end-to-end congestion control negatively.
+
```
Comment (abrunstrom) [inline draft-quic-atsss-reqs.mkd:473]: This is quite speculative. What the end-to-end latency is can vary and is in general not known. I would not claim that the cc control loops will not interact.

**Pull request merged**